pkg/netpol/diff/diff_test.go

```diff
@@ -172,171 +172,204 @@ func TestDiff(t *testing.T) {
+		_, err1 = diffAnalyzer.ConnectivityDiffToString(connsDiff1)
+		_, err2 = diffAnalyzerStopsOnError.ConnectivityDiffToString(connsDiff2)
+		diffErrors1 := diffAnalyzer.Errors()
+		require.Equal(t, err1.Error(), entry.firstErrStr, "test: %s", entry.name)
+		require.Equal(t, err2.Error(), entry.firstErrStr, "test: %s", entry.name)
+		require.True(t, errors.As(diffErrors1[0].Error(), &formattingErrType), "test: %s", entry.name)
+	}
+}
+
+func TestSevereErrors(t *testing.T) {
+	// testing behavior with severe error, analyzer without stopOnError will continue running regularly,
+	// analyzer with stopOnError will stop on first severe error and return empty result
+	cases := []testErrEntry{
+		{
+			// description: only first dir has severe error ,
+			// it also has a warning which was captured before the severe error so expected to appear in both
+			name:                           "dir 1 has no k8s resources",
+			dir1:                           filepath.Join("bad_yamls", "not_a_k8s_resource.yaml"),
+			dir2:                           "ipblockstest", // no warnings, nor any severe/fatal errors
+			firstErrStr:                    "Yaml document is not a K8s resource",
+			expectedErrNumWithoutStopOnErr: 2,
+			expectedErrNumWithStopOnErr:    2,
 		},
 		{
-			name:         "dir 1 does not exists",
-			dir1:         filepath.Join("bad_yamls", "subdir3"),
-			dir2:         "ipblockstest",
-			errStr:       "error accessing directory:",
-			isCaFatalErr: true,
+			// description: only first dir has severe error , it also has a warning
+			// the severe error is captured first, so we expect not to see the warning when running with stopOnError as it stops running
+			name:                           "dir 1 has malformed yaml",
+			dir1:                           filepath.Join("bad_yamls", "document_with_syntax_error.yaml"),
+			dir2:                           "ipblockstest", // no warnings, nor any severe/fatal errors
+			firstErrStr:                    "YAML document is malformed",
+			expectedErrNumWithoutStopOnErr: 2,
+			expectedErrNumWithStopOnErr:    1,
 		},
 		{
-			name:          "dir 1 has no k8s resources",
-			dir1:          filepath.Join("bad_yamls", "not_a_k8s_resource.yaml"),
-			dir2:          "ipblockstest",
-			errStr:        "Yaml document is not a K8s resource",
-			isCaSevereErr: true, // severe error, stops only if stopOnError = true
+			// dirty directory, includes 3 severe errors
+			// when running without stopOnError we expect to see 6 severe errors (3 for each dir flag)
+			// but when running with stopOnError we expect to see only 1 , and then stops
+			name:                           "both dirs return severe errors on their malformed yaml files",
+			dir1:                           "dirty",
+			dir2:                           "dirty",
+			firstErrStr:                    "YAML document is malformed",
+			expectedErrNumWithoutStopOnErr: 6,
+			expectedErrNumWithStopOnErr:    1,
 		},
```

Comment:
this is another example,
after the fix we see `expectedErrNumWithoutStopOnErr: 6,`  (as `dirty` dir has 3 errors , so for dir1+dir2 it is 6)
before the fix we would get  `expectedErrNumWithoutStopOnErr: 9,`